frontend/src/components/Upgrade.tsx

```diff
@@ -1,20 +1,60 @@
 import { IUpgrade, UpgradeStatus } from "../../../shared/types";
+import { RGBTripleToCSS } from "../../../shared/util";
+import UpgradeImage from "./UpgradeImage";
 
 interface IUpgradeProps {
   up: IUpgrade;
   status: UpgradeStatus;
   onBuy: (up: IUpgrade) => void;
+  currencyAmount: number;
 }
 
-const Upgrade = ({ up, status, onBuy }: IUpgradeProps) => {
-  return (
-    <div
-      className={`upgrade status-${status}`}
-      onClick={() => {
+const bgUnavailableRGBT = { r: 179, g: 176, b: 176 };
```

Comment:
Because you created the RGB triple interface, you can set those types here, ex:

```ts
const bgUnavailableRGBT: RGBTriple = { r: 179, g: 176, b: 176 };
```